lib/temporal/worker.rb

```diff
@@ -0,0 +1,72 @@
+require 'async'
+require 'temporal/bridge'
+require 'temporal/data_converter'
+require 'temporal/failure_converter'
+require 'temporal/payload_converter'
+require 'temporal/runtime'
+require 'temporal/worker/activity'
+require 'temporal/worker/thread_pool_executor'
+
+module Temporal
+  class Worker
+    # TODO: Add worker interceptors
+    def initialize(
+      connection,
+      namespace,
```

Comment:
It is an unfortunate side effect to the connection model that someone writing code to start a worker has to know the namespace (in TS we had to because it doesn't share connections, but luckily everywhere else we don't have to and therefore don't burden the user with it). It is also unfortunate that they have to know to pass the same converter values.

Since continuing to not want to let a worker use a client here, can we at least have some kind of helper to create a worker from a client using its namespace and data conversion configuration? (and identity and combined client/worker interceptors in the future)

Also, there is talk of allowing activities access to the Temporal client they are created with, so we might need to factor that in.

Reply:
Yes, this is a known drawback and I think we still have some differences in how this should work. I agree that some way to partially share config between a Client and a Worker is needed to make this nicer. But I'm not convinced it should be the Client. Maybe we can instead create a composable Config class that both Client and Worker can be initialized with. Or maybe the Runtime can play this role as both Client and Worker depend on it.

I'll think of possible solutions here and let's discuss these outside of this PR.